pkg/controllers/netpol/namespace.go

```diff
@@ -0,0 +1,69 @@
+package netpol
+
+import (
+	"github.com/golang/glog"
+	api "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (npc *NetworkPolicyController) newNamespaceEventHandler() cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			npc.handleNamespaceAdd(obj.(*api.Namespace))
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			npc.handleNamespaceUpdate(oldObj.(*api.Namespace), newObj.(*api.Namespace))
+		},
+		DeleteFunc: func(obj interface{}) {
+			switch obj := obj.(type) {
+			case *api.Namespace:
+				npc.handleNamespaceDelete(obj)
+				return
+			case cache.DeletedFinalStateUnknown:
+				npc.handleNamespaceDelete(obj.Obj.(*api.Namespace))
```

Comment:
We still need to check if obj.Obj can be type casted to `*api.Namespace`

Reply:
done